runtime/tests/staking_reward.rs

```diff
@@ -549,3 +550,50 @@ fn authorship_reward_of_a_chilled_validator() {
 			);
 		});
 }
+
+#[test]
+/// This tests if slash goes to treasury as CENNZ.
+fn slashed_cennz_gets_into_treasury() {
+	let validators = make_authority_keys(6);
+	let initial_balance = 1_000 * DOLLARS;
+
+	ExtBuilder::default()
+		.slash_defer_duration(0)
+		.initial_authorities(validators.as_slice())
+		.initial_balance(initial_balance)
+		.stash(initial_balance)
+		.build()
+		.execute_with(|| {
+			let cennz_amount = 10_000;
+			let cpay_amount = 12_112;
+			let updated_cennz_amount = 4_210_000;
+			let update_rewarded_cpay_amount = 12_449;
+			let _ = <GenericAsset as MultiCurrencyAccounting>::deposit_creating(
+				&Treasury::account_id(),
+				Some(STAKING_ASSET_ID),
+				cennz_amount,
+			);
+			let _ = <GenericAsset as MultiCurrencyAccounting>::deposit_creating(
+				&Treasury::account_id(),
+				Some(SPENDING_ASSET_ID),
+				cpay_amount,
+			);
+			assert_eq!(Staking::current_era(), 0);
```

Comment:
You need to check the treasury balances here so we know what it was before and what it became after the slashing.